Mathlib/NumberTheory/Divisors.lean

```diff
@@ -514,3 +520,65 @@ theorem disjoint_divisors_filter_isPrimePow {a b : ℕ} (hab : a.Coprime b) :
   exact hn.ne_one (Nat.eq_one_of_dvd_coprimes hab han hbn)
 
 end Nat
+
+namespace Int
+variable {z : ℤ} {x : ℤ × ℤ}
+
+local notation "natCast" => Nat.castEmbedding (R := ℤ)
+local notation "negNatCast" =>
+  Function.Embedding.trans Nat.castEmbedding (Equiv.toEmbedding (Equiv.neg ℤ))
+
+/-- Pairs of divisors of an integer as a finset.
+
+`z.divisorsAntidiag` is the finset of pairs `(a, b) : ℤ × ℤ` such that `a * b = z`.
+As a special case, `Int.divisorsAntidiag 0 = ∅`.
+
+O(|z|). Computed from `Nat.divisorsAntidiagonal`. -/
+def divisorsAntidiag : (z : ℤ) → Finset (ℤ × ℤ)
+  | (n : ℕ) =>
+    let s : Finset (ℕ × ℕ) := n.divisorsAntidiagonal
+    (s.map <| .prodMap natCast natCast).disjUnion (s.map <| .prodMap negNatCast negNatCast) <| by
+      simp +contextual [s, disjoint_left, eq_comm]
+  | negSucc n =>
+    let s : Finset (ℕ × ℕ) := (n + 1).divisorsAntidiagonal
+    (s.map <| .prodMap natCast negNatCast).disjUnion (s.map <| .prodMap negNatCast natCast) <| by
+      simp +contextual [s, disjoint_left, eq_comm, forall_swap (α := _ * _ = _)]
+
+@[simp]
+lemma mem_divisorsAntidiag : ∀ {z} {x : ℤ × ℤ}, x ∈ divisorsAntidiag z ↔ x.fst * x.snd = z ∧ z ≠ 0
+  | (n : ℕ), ((a : ℕ), (b : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    simp +contextual [eq_comm]
+  | (n : ℕ), (negSucc a, negSucc b) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+  | (n : ℕ), ((a : ℕ), negSucc b) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    aesop
+  | (n : ℕ), (negSucc a, (b : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    aesop
+  | .negSucc n, ((a : ℕ), (b : ℕ)) => by
+    simp [divisorsAntidiag]
+    norm_cast
+    aesop
+  | .negSucc n, (negSucc a, negSucc b) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+  | .negSucc n, ((a : ℕ), negSucc b) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    aesop
+  | .negSucc n, (negSucc a, (b : ℕ)) => by
+    simp [divisorsAntidiag, negSucc_eq, -neg_add_rev]
+    norm_cast
+    simp +contextual [eq_comm]
+
+@[simp] lemma divisorsAntidiag_zero : divisorsAntidiag 0 = ∅ := by simp [divisorsAntidiag]
+
```

Comment:
Could you prove `swap_mem_divisorsAntidiagonal` and `map_swap_divisorsAntidiagonal` as well?